mne/io/brainvision/brainvision.py

```diff
@@ -136,12 +139,44 @@ def __init__(self, vhdr_fname, montage=None,
                 offsets = None
                 n_samples = n_samples // (dtype_bytes * (info['nchan'] - 1))
         self.preload = False  # so the event-setting works
+
+        # Get annotations from vmrk file and then use events_from_annotations
+        trig_shift_by_type = _check_trig_shift_by_type(trig_shift_by_type)
+        annots = read_annotations_brainvision(mrk_fname, info['sfreq'])
+        event_id = dict() if event_id is None else event_id
+        dropped_desc = []  # use to collect dropped descriptions
+        event_id = partial(_event_id_func, event_id=event_id,
+                           trig_shift_by_type=trig_shift_by_type,
+                           dropped_desc=dropped_desc)
+
+        # XXX : We use a mock raw as self is not proper yet
+        # and it's not possible to do it after as one needs the events
+        # to create the stim channel before calling super init
+        raw_tmp = _RawShell()
+        raw_tmp.first_samp = 0
+        raw_tmp.info = info
+        raw_tmp.set_annotations(annots)
+
+        def _time_as_index(times, use_rounding, origin):
+            return np.round(np.atleast_1d(times) * info['sfreq'])
+
+        raw_tmp.time_as_index = _time_as_index
+
+        on_drop = 'ignore'   # ignore here as warning raised below
+        events, _ = events_from_annotations(raw_tmp, event_id, on_drop=on_drop)
+
+        _report_dropped(dropped_desc, on_drop='warn')
```

Comment:
`on_drop=on_drop`

Reply:
Ok, now I get it. 

```
events, _ = events_from_annotations(raw_tmp, event_id)
_report_dropped(dropped_desc, on_drop='warn')
```
and remove it from `events_from_annotations` we don't use it anywhere else.